src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -1296,19 +1294,32 @@ pub fn compare_const_vals<'a, 'tcx>(
 }
 
 #[derive(PartialEq)]
-enum LitToConstError {
+pub enum LitToConstError {
     UnparseableFloat,
-    Propagated,
+    Reported,
 }
 
-// FIXME: Combine with rustc_mir::hair::cx::const_eval_literal
-fn lit_to_const<'a, 'tcx>(lit: &'tcx ast::LitKind,
-                          tcx: TyCtxt<'a, 'tcx, 'tcx>,
-                          ty: Ty<'tcx>,
-                          neg: bool)
-                          -> Result<&'tcx ty::Const<'tcx>, LitToConstError> {
+pub fn lit_to_const<'a, 'gcx, 'tcx>(
+    lit: &'tcx ast::LitKind,
+    tcx: TyCtxt<'a, 'gcx, 'tcx>,
+    ty: Ty<'tcx>,
+    neg: bool,
+) -> Result<&'tcx ty::Const<'tcx>, LitToConstError> {
```

Comment:
Can this be moved out of `hair::pattern`? Somewhere else in `hair` would be fine (cc @nikomatsakis).

Reply:
I created a custom file, there's likely other things we'll want to add in the future.